livekit-plugins/livekit-plugins-speechify/livekit/plugins/speechify/tts.py

```diff
@@ -0,0 +1,293 @@
+
+    def synthesize(
+        self,
+        text: str,
+        *,
+        conn_options: APIConnectOptions = DEFAULT_API_CONNECT_OPTIONS,
+    ) -> ChunkedStream:
+        return ChunkedStream(
+            tts=self,
+            input_text=text,
+            conn_options=conn_options,
+            opts=self._opts,
+            session=self._ensure_session(),
+        )
+
+
+class ChunkedStream(tts.ChunkedStream):
+    """Synthesize using the chunked api endpoint"""
+
+    def __init__(
+        self,
+        *,
+        tts: TTS,
+        input_text: str,
+        opts: _TTSOptions,
+        conn_options: APIConnectOptions,
+        session: aiohttp.ClientSession,
+    ) -> None:
+        super().__init__(tts=tts, input_text=input_text, conn_options=conn_options)
+        self._opts, self._session = opts, session
+
+    async def _run(self) -> None:
+        request_id = utils.shortuuid()
+        data = {
+            "input": self._input_text,
+            "voice_id": self._opts.voice_id,
+            "language": self._opts.language if is_given(self._opts.language) else None,
+            "model": self._opts.model if is_given(self._opts.model) else None,
+            "audio_format": _audio_format_from_encoding(self._opts.encoding),
+            "options": {
+                "loudness_normalization": self._opts.loudness_normalization
+                if is_given(self._opts.loudness_normalization)
+                else None,
+                "text_normalization": self._opts.text_normalization
+                if is_given(self._opts.text_normalization)
+                else None,
+            },
+        }
+
+        decoder = utils.codecs.AudioStreamDecoder(
+            sample_rate=self._opts.sample_rate,
+            num_channels=1,
+        )
+
+        decode_task: asyncio.Task | None = None
+        try:
+            async with self._session.post(
+                _synthesize_url(self._opts),
+                headers=_get_auth_header(self._opts.token),
+                json=data,
+                timeout=self._conn_options.timeout,
+            ) as resp:
+                if not resp.content_type.startswith("audio/"):
+                    content = await resp.text()
+                    logger.error("11labs returned non-audio data: %s", content)
+                    return
+
+                async def _decode_loop():
+                    try:
+                        async for bytes_data, _ in resp.content.iter_chunks():
+                            decoder.push(bytes_data)
+                    finally:
+                        decoder.end_input()
+
+                decode_task = asyncio.create_task(_decode_loop())
+                emitter = tts.SynthesizedAudioEmitter(
+                    event_ch=self._event_ch,
+                    request_id=request_id,
+                )
+                async for frame in decoder:
+                    emitter.push(frame)
+                emitter.flush()
```

Comment:
```suggestion
                emitter.flush()
                
                await decode_task
```
awaiting here in case an error happened inside _decode_loop, it'll be propagated correctly